melodic-manatees/early_internet/background_app/views.py

```diff
@@ -0,0 +1,107 @@
+from .forms import FileUploadForm
+from .models import BackgroundFile
+from users.models import UserProfile
+from . import background_utility
+import os
+from django.core.files import File
+from django.shortcuts import render, redirect
+from django.contrib import messages
+from django.contrib.auth.decorators import login_required
+
+'''All background related views go here'''
+
+
+def background_home(request):
+    # get all background objects associated with this user,
+    # and pass them as context to the HTML page
+    background_list = BackgroundFile.objects.filter(
+        background_owner=request.user.profile
+    )
+    print(len(background_list))
+    return render(
+        request,
+        'background_app/background_home.html',
+        {'background_list': background_list}
+    )
+
+
+@login_required(login_url='login')
+def add_background(request):
+    if request.method == 'POST':
```

Comment:
For views that have to handle more than one HTTP method, it's a great idea to use class-based views instead of function-based views. With class-based views, you can handle each HTTP method in a separate function, which makes it easier to write very clear and readable code.

For example, I think this view makes it a little too implicit which path handles the GET requests, since you're not even explicitely writing GET anywhere, it's just caught by the else clause. This view will also erronously handle HTTP methods that don't make any sense, like PATCH or DELETE, which feels very imprecise.